### PR TITLE
14 extract examples to separate repository

### PR DESCRIPTION
**Pull request by lukaswajdzik**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>nhood-examples</artifactId>
+    <version>0.0.2</version>
+
+	<parent>
```
[the remark this answers is not in the document]
Comment (lukaswajdzik) [reply pom.xml]: done

File: README.md [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+
+- Java 8
+- Maven
+
+## Docker
+
+To run the examples in docker, call the command from nhood-examples folder:
+
+```bash
+docker run -d -P  -v {/path/to/.m2}:/root/.m2 -v `pwd`:/test_app maven:3.6-jdk-11 cd /test_app && mvn clean install
```
[the remark this answers is not in the document]
Comment (lukaswajdzik) [reply README.md]: done

**Comment by krzysztofreczek** [comment]
😍

File: LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (krzysztofreczek) [inline LICENSE.md:1]: This file does need *.md extension. Just call it LICENSE

File: README.md [hunk 4]
```diff
@@ -1,2 +1,48 @@
...
+```
+
+## CI/CD
+
+Project is continuously integrated with `circleCi` pipeline that link to which may be found [here](https://circleci.com/gh/nhood-org/workflows/nhood-engine)
+
+Pipeline is fairly simple:
+
+1. Build and test project with a set of jdk: `1.8` and `11`.
+2. Deploy new snapshot version to [nhood maven repository](https://github.com/nhood-org/nhood-repository/tree/mvn-repo/com/h8/nh)
```
Comment (krzysztofreczek) [inline README.md:41]: This is not true, we do not deploy anything as of now.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,48 @@
...
+
+In order to build and run the tests use the command:
+
+```bash
+mvn clean install
+```
+
+## CI/CD
+
+Project is continuously integrated with `circleCi` pipeline that link to which may be found [here](https://circleci.com/gh/nhood-org/workflows/nhood-engine)
```
Comment (krzysztofreczek) [inline README.md:36]: Should be https://circleci.com/gh/nhood-org/workflows/nhood-engine-examples

File: README.md [hunk 6]
```diff
@@ -1,2 +1,48 @@
...
+Pipeline is fairly simple:
+
+1. Build and test project with a set of jdk: `1.8` and `11`.
+2. Deploy new snapshot version to [nhood maven repository](https://github.com/nhood-org/nhood-repository/tree/mvn-repo/com/h8/nh)
+
+Configuration of CI is implemented in `.circleci` and  `.circleci.setting.xml`.
+
+## License
+
+`nhood-engine` is released under the MIT license:
```
Comment (krzysztofreczek) [inline README.md:47]: nhood-engine-examples

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            <artifactId>opencsv</artifactId>
+            <version>4.3.2</version>
+        </dependency>
+
+        <!--Test-->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>${junit.jupiter.version}</version>
+
```
Comment (krzysztofreczek) [inline pom.xml:77]: Mind formatting :)

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+			<id>github</id>
+			<url>https://maven.pkg.github.com/nhood-org/repository</url>
+		</repository>
+	</repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.h8.nh</groupId>
+            <artifactId>nhood-engine-core</artifactId>
+            <version>0.0.3</version>
```
Comment (krzysztofreczek) [inline pom.xml:39]: those nhood versions may be defined as propery

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.h8.nh</groupId>
+    <artifactId>nhood-examples</artifactId>
+    <version>0.0.3</version>
```
Comment (krzysztofreczek) [inline pom.xml:7]: We can start independent versioning of examples, let's go with 0.0.1

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+        <junit.jupiter.version>5.6.0</junit.jupiter.version>
+        <junit.platform.version>1.6.0</junit.platform.version>
+        <assertj.version>3.15.0</assertj.version>
+        <mockito.version>3.3.0</mockito.version>
+        <surfire.plugin.version>3.0.0-M4</surfire.plugin.version>
+        <!--Servers-->
+        <github.global.server>github</github.global.server>
+        <project.scm.id>github</project.scm.id>
+
```
Comment (krzysztofreczek) [inline pom.xml:25]: Mind formatting, pls

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+            <version>${assertj.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>${mockito.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (krzysztofreczek) [inline pom.xml:110]: Please make sure a whole file is formatted coherently

File: LICENSE.md (hunk 3 above)
Comment (lukaswajdzik) [reply LICENSE.md:1]: done

File: README.md (hunk 4 above)
Comment (lukaswajdzik) [reply README.md:41]: removed

File: README.md (hunk 5 above)
Comment (lukaswajdzik) [reply README.md:36]: done

File: README.md (hunk 6 above)
Comment (lukaswajdzik) [reply README.md:47]: done

File: pom.xml (hunk 7 above)
Comment (lukaswajdzik) [reply pom.xml:77]: done

File: pom.xml (hunk 8 above)
Comment (lukaswajdzik) [reply pom.xml:39]: done

File: pom.xml (hunk 9 above)
Comment (lukaswajdzik) [reply pom.xml:7]: done

File: pom.xml (hunk 10 above)
Comment (lukaswajdzik) [reply pom.xml:25]: done

File: pom.xml (hunk 11 above)
Comment (lukaswajdzik) [reply pom.xml:110]: Do you use any formatter tool ?

Comment (krzysztofreczek) [reply pom.xml:114]: Not for xmls, I believe intelliJ may help here.

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+        <junit.platform.version>1.6.0</junit.platform.version>
+        <assertj.version>3.15.0</assertj.version>
+        <mockito.version>3.3.0</mockito.version>
+        <surfire.plugin.version>3.0.0-M4</surfire.plugin.version>
+        <!--Servers-->
+        <github.global.server>github</github.global.server>
+        <project.scm.id>github</project.scm.id>
+
+        <nhood.version>0.0.3</nhood.version>
+        <checkstyle.plugin.version>3.1.1</checkstyle.plugin.version>
```
Comment (krzysztofreczek) [inline pom.xml:28]:
```suggestion
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <java.version>1.8</java.version>
        
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        
        <nhood.version>0.0.3</nhood.version>

        <!--test-->
        <junit.jupiter.version>5.6.0</junit.jupiter.version>
        <junit.platform.version>1.6.0</junit.platform.version>
        <assertj.version>3.15.0</assertj.version>
        <mockito.version>3.3.0</mockito.version>
        
        <!--plugins-->
        <surfire.plugin.version>3.0.0-M4</surfire.plugin.version>
        <checkstyle.plugin.version>3.1.1</checkstyle.plugin.version>
        
        <!--repository-->
        <github.global.server>github</github.global.server>
        <project.scm.id>github</project.scm.id>
```

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.h8.nh</groupId>
+    <artifactId>nhood-examples</artifactId>
+    <version>0.0.1</version>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (krzysztofreczek) [inline pom.xml:11]: I just reorganized properties a little bit.

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <java.version>1.8</java.version>
+        
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        
+        <nhood.version>0.0.3</nhood.version>
```
Comment (krzysztofreczek) [inline pom.xml:18]:
```suggestion
        <nhood-engine.version>0.0.3</nhood.version>
```

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+        <dependency>
+            <groupId>com.h8.nh</groupId>
+            <artifactId>nhood-engine-matrix</artifactId>
+            <version>${nhood.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.h8.nh</groupId>
+            <artifactId>nhood-engine-test-utils</artifactId>
+            <version>${nhood.version}</version>
```
Comment (krzysztofreczek) [inline pom.xml:70]:
```suggestion
        <dependency>
            <groupId>com.h8.nh</groupId>
            <artifactId>nhood-engine-core</artifactId>
            <version>${nhood-engine.version}</version>
        </dependency>

        <dependency>
            <groupId>com.h8.nh</groupId>
            <artifactId>nhood-engine-core-api</artifactId>
            <version>${nhood-engine.version}</version>
        </dependency>

        <dependency>
            <groupId>com.h8.nh</groupId>
            <artifactId>nhood-engine-matrix-api</artifactId>
            <version>${nhood-engine.version}</version>
        </dependency>

        <dependency>
            <groupId>com.h8.nh</groupId>
            <artifactId>nhood-engine-matrix</artifactId>
            <version>${nhood-engine.version}</version>
        </dependency>

        <dependency>
            <groupId>com.h8.nh</groupId>
            <artifactId>nhood-engine-test-utils</artifactId>
            <version>${nhood-engine.version}</version>
```

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <java.version>1.8</java.version>
+        
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        
+        <nhood-engine.version>0.0.3</nhood.version>
```
Comment (krzysztofreczek) [inline pom.xml:18]:
```suggestion
        <nhood-engine.version>0.0.3</nhood-engine.version>
```

**Comment by krzysztofreczek** [comment]
Feel free to merge it! Well done! Please squash all commit when merging

**Comment by krzysztofreczek** [comment]
PR implements https://github.com/nhood-org/nhood-examples/issues/4

**Comment by krzysztofreczek** [comment]
PR partially implements https://github.com/nhood-org/nhood-engine/issues/14

**Pull request merged**